collector/mscluster_network.go

```diff
@@ -0,0 +1,116 @@
+package collector
+
+import (
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	registerCollector("mscluster_network", newMSCluster_NetworkCollector) // TODO: Add any perflib dependencies here
+}
+
+// A MSCluster_NetworkCollector is a Prometheus collector for WMI MSCluster_Network metrics
+type MSCluster_NetworkCollector struct {
+	Characteristics *prometheus.Desc
+	Flags           *prometheus.Desc
+	Metric          *prometheus.Desc
+	Role            *prometheus.Desc
+	State           *prometheus.Desc
+}
+
+func newMSCluster_NetworkCollector() (Collector, error) {
+	const subsystem = "mscluster_network"
+	return &MSCluster_NetworkCollector{
+		Characteristics: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "characteristics"),
+			"(Characteristics)",
+			[]string{"name"},
+			nil,
+		),
+		Flags: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "flags"),
+			"(Flags)",
+			[]string{"name"},
+			nil,
+		),
+		Metric: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "metric"),
+			"(Metric)",
+			[]string{"name"},
+			nil,
+		),
+		Role: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "role"),
+			"(Role)",
+			[]string{"name"},
+			nil,
+		),
+		State: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "state"),
+			"(State)",
+			[]string{"name"},
+			nil,
+		),
```

Comment:
I'd be keen to see additional detail in the metric description here, and under the `docs/` directory. It will be difficult for users to determine the purpose of a metric without a description or further documentation.

Reply:
I've updated all the descriptions and added the docs.